client/dialer.go

```diff
@@ -55,31 +53,18 @@ func (c *HostClient) makeDialer(superProxy *superproxy.SuperProxy,
 	//set https tls config
 	switch reqType {
 	case requestDirectHTTP:
-		return dialerWrapper(transport.Dial(targetWithPort))
+		fallthrough
 	case requestDirectHTTPS:
-		if c.tlsServerConfig == nil {
-			c.tlsServerConfig = cert.MakeClientTLSConfig("", targetTLSServerName)
-		}
-		return dialerWrapper(transport.DialTLS(targetWithPort, c.tlsServerConfig))
+		return dialerWrapper(transport.Dial(targetWithPort))
 	case requestProxyHTTP:
 		return dialerWrapper(transport.Dial(superProxy.HostWithPort()))
 	case requestProxyHTTPS:
```

Comment:
it is also the dialer's responsibility to make a TLS dialer for secure super proxy, for any client or raw tunnel purpose